README-zh_CN.md

```diff
@@ -0,0 +1,430 @@
+
+</details>
+
+## 外置框架
+
+<details>
+<summary>不要将 React、Vue 等框架打包到你的库中</summary>
+
+当构建的库依赖某个框架（例如 React、Vue 等），或者是作为另一个库的插件，你可能需要将框架配置到“externals”中。这可以使你的库引用这个框架，但不会将其包含在最终的产出中。这会防止产生一些 bug，并减少库的体积。
+
+你应该还需要将框架添加到库的 `package.json` 的 [peer dependencies](#列出-peerdependencies) 中，这将帮助开发者发现你依赖于某个框架。
+
+</details>
+
+## 面向现代浏览器
+
+<details>
+<summary>如果需要，使用新特性，让开发者支持旧浏览器</summary>
+
+[这篇 web.dev 上的文章](https://web.dev/publish-modern-javascript/)提供了一个很好的案例，并提供了关于指导原则：
+
+- 当使用你的库时，允许开发者支持老版本的浏览器。
+- 输出多个产出来支持不同版本的浏览器。
+
+举个例子，如果你使用 TypeScript，你应该在 `tsconfig.json` 中将 `"target"` 设置为 `ESNext`。
+
+</details>
+
+## 必要的编译
+
+<details>
+<summary>编译 TypeScript、将 JSX 转换为函数调用</summary>
+
+如果库的源码是需要进行编译的形式，如 TypeScript、React 或 Vue 组件等，那么你库需要输出的是编译后的代码。
+
+例如：
+
+- 你的 TypeScript 代码应该输出为 JavaScript。
+- 你的 React 组件，例如 `<Example />`，应该在输出中使用 `jsx()` 或 `createElement()` 来替换 JSX 语法。
+
+进行这样的编译时，请确保同时也[创建 sourcemap](#创建-sourcemap)
+
+</details>
+
+## 维护 changelog
+
+<details>
+<summary>追踪更新和变更</summary>
+
+只要能让开发者了解到有哪些变更和对他们的影响，至于是通过自动化工具还是通过亲自动手的方式来处理，这都无关紧要。理想情况下，库的每次[版本](#设置-version)变更都应该在 changelog 中进行相应的更新。
+
+</details>
+
+## 配置 `package.json`
+
+`package.json` 中有许多重要的配置字段值得讨论；我在这里将着重讨论其中最为重要的一些，这还有很多 [additional fields](https://docs.npmjs.com/cli/v8/configuring-npm/package-json)，你同样可以进行配置。
+
+### 设置 `name`
+
+<details>
+<summary>给你的库取一个名字</summary>
+
+`name` 字段将决定你的包在 `npm` 上的名字，开发者可以通过这个名字去安装并使用你的库。
+
+注意，库的命名是有限制的，如果你的代码库属于某个组织，你还可以创建一个命名空间。更多细节可以参考 [name docs on npm](https://docs.npmjs.com/cli/v8/configuring-npm/)。
+
+`name` 和 [version](#设置-version) 的组合为库每次迭代创建一个唯一标识。
+
+</details>
+
+### 设置 `version`
+
+<details>
+<summary>通过更改 version 来对你的库发布更新</summary>
+
+正如 [name](#设置-name) 部分所说，`name` 和 `version` 的组合为你的库在 npm 上创建一个唯一标识。当你更新库中的代码时，你可以更新 `version` 字段并发布以允许开发者获取该新代码。
+
+推荐使用 [semver](https://semver.org/) 版本控制策略，但要注意的是有些库选择 [calver](https://calver.org/) 或使用他们自己特有的版本控制策略。无论你选择使用哪种策略，都应该记录下来，以便开发者了解你的库是如何进行版本控制的。
+
+你还应该在 [changelog](#维护-changelog) 中记录你的更改。
+
+</details>
+
+### 定义你的 `exports`
+
+<details>
+<summary><code>exports</code> 为你的库定义公共 API</summary>
+
+`package.json` 中的 `exports` 字段 - 有时被称为“导出映射” - 是一个非常有用的补充，尽管它确实引入了一些复杂性。它做的最重要的两件事是：
+
+1. 定义哪些东西可以从你的库中导入，哪些则不可以，以及可导入的内容的名字。如果没有在 `exports` 中被列出，那么开发者就不可以 `import` 或 `require` 它们。换句话说，`exports` 的表现像是给你的库用户查看的公共 API，帮助定义哪些是外部的哪些是内部的。
+
+2. 允许你根据不同的条件（你可以定义）去选择那个文件是被导入的，例如“文件是被 `import` 还是被 `require`？开发人员需要的是 `development` 版本的库还是 `production` 版本等等。
+
+关于这部分的内容[NodeJS 团队](https://nodejs.org/api/packages.html#package-entry-points)和[Webpack 团队](https://webpack.js.org/guides/package-exports/)提供了一些很优秀的文档。在此我们就列出一个涵盖大部分常用场景的例子：
+
+```json
+{
+  "exports": {
+    ".": {
+      "types": "index.d.ts",
+      "script": "index.umd.js",
+      "module": "index.js",
+      "import": "index.js",
+      "require": "index.cjs",
+      "default": "index.js"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+</details>
```

Comment:
I believe this needs to be removed; when previewing your doc, the content below is shown without being behind the toggle.

Reply:
yes, I will fix this.